minetest/minetest
# Formspec: Allow multiple mouse button click for touch-screen

**Goal of the PR**
This PR tries to restore old touch behaviour on inventory management.

https://github.com/minetest/minetest/assets/4017302/17c35f2d-a125-45fd-9b86-2da66c207aea

**How does the PR work?**
The current code prohibits multiple button click/press action in a single mouse, but the behaviour is different for touch-screen devices because mouse clicks are simulated from touches. This PR removes that limit only for touch-screen devices.

**Does it resolve any reported issue?**
Yes, this PR tries to fix #13822.

**Does this relate to a goal in [the roadmap](https://github.com/minetest/minetest/blob/master/doc/direction.md)?**
Yes, this PR fixes a regression.

## To do

This PR is Ready for Review.

## How to test

1. Play a Minetest world in a touch-screen device.
2. Open the inventory dialog.
3. Tap and hold a stack of item.
4. Do 2nd tap using different finger/hand (different touch).
5. A single item should be placed on the first touch's position.
6. Move the first touch to another slot.
7. Do 2nd tap again (like step number 4).
8. Another single item should be placed on the new first touch's position.
9. Play a Minetest world in a non-touch-screen device.
10. All inventory-related action should work as usual.
